src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/WhatIfShortCircuitingRuleTests.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Rules;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
+{
+    [TestClass]
+    public class WhatIfShortCircuitingRuleTests : LinterRuleTestsBase
+    {
+        private static readonly ServiceBuilder Services = new ServiceBuilder()
+            .WithRegistration(x => x.AddSingleton(
+                IConfigurationManager.WithStaticConfiguration(
+                    IConfigurationManager.GetBuiltInConfiguration()
+                    .WithAllAnalyzers())));
+
+        private readonly string SAModuleContent = @"
```

Comment:
[nit]: Use triple-quoted multiline strings `"""..."""` instead of `@`-prefixed multiline strings when the lefthand margin of the string and the lefthand margin of the containing file don't align.

_passim_